### PR TITLE
Added initial generators qunit and newqunittest

### PR DESCRIPTION
**Pull request by georgbischoff**

### CODE REVIEW

File: generators/newqunittest/templates/test/unit/$testFile.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		});<% } %>		
+		<% if (useSinonJS) { %>QUnit.test("<%= testName %>", function (assert) {
+			//let spy = sandbox.spy("<placeholder>", "<placeholder>");
+			//let stub = sandbox.stub("<placeholder>", "<placeholder>").returns(...);
+			//let var mock = sandbox.mock(<placeholder);
+			//https://sinonjs.org/
+			//assert.equal(<actual result>, <expected result>, <optional message>);
+			//assert.ok(<actual result>, <optional message>);
+			assert.ok(false, "Implement me");
+		});<% } %>
```
Comment (ArnaudBuchholz) [inline generators/newqunittest/templates/test/unit/$testFile.js:30]: Would that makes sense to also insert an example of asynchronous test ?

File: generators/qunit/templates/test/testsuite.qunit.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+window.suite = function () {
+    "use strict";
+
+    var oSuite = new parent.jsUnitTestSuite(),
+        sContextPath = location.pathname.substring(0, location.pathname.lastIndexOf("/") + 1);
+        
+    oSuite.addTestPage(sContextPath + "unit/unitTests.qunit.html");
+    <% if (typeof(opa5Journeys) !== "undefined") { %>oSuite.addTestPage(sContextPath + "integration/opaTests.qunit.html");
```
Comment (ArnaudBuchholz) [inline generators/qunit/templates/test/testsuite.qunit.js:8]: Would that make sense to apply the journey splitting pattern ?

File: test/qunit.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+const assert = require("yeoman-assert");
```
Comment (ArnaudBuchholz) [inline test/qunit.js:1]: Nice that you can test the generator

**Review by ArnaudBuchholz — COMMENTED** [review]
I am not very familiar with yeoman generators and it is hard to read through the proposed options.
I just noted two possible improvements.

File: generators/newqunittest/templates/test/unit/$testFile.js (hunk 1 above)
Comment (georgbischoff) [reply generators/newqunittest/templates/test/unit/$testFile.js:30]: Great idea. I added two options for the testing of asynchronous operations and errorThrows

**Comment by IObert** [comment]
Thanks for your PR and the work you put in it. I just tested the generator and noticed the following things:

* I wanted to run project:newqunittest first as I assumed that all files needed for qunit tests are already present. And I was able to run the subgenerator without any errors but I noticed that it didn't create useful files:
Screenshot 2021-08-18 at 15 29 03
 => I think it'd be nice if the subgenerator wouldn't fail if the needed files to run it aren't there


* Then, I ran  project:qunit in a new project which added a few more files. Once the files were generated, I ran `npm test` and noticed that the new files break the following es-lint rules:
Screenshot 2021-08-18 at 15 46 56

=> Can you please adjust the files so that the test completes succcessfully

File: generators/qunit/templates/test/testsuite.qunit.js (hunk 2 above)
Comment (georgbischoff) [reply generators/qunit/templates/test/testsuite.qunit.js:8]: We can discuss that for the opa-generator.

**Comment by georgbischoff** [comment]
- Fixed the es-lint issues
- Added selection uimodules to generate single qunit test in the right folder. Also added a remark to run the qunit generator if the overall test-suite was not yet generated.

**Comment by IObert** [comment]
Thanks a lot!

**Pull request merged**